### PR TITLE
Disable the constant folding pass in Rust < 1.26.

### PR DESCRIPTION
**Pull request by sunfishcode**
The constant folding pass depends on rustc_apfloat which uses i128,
which is new as of Rust 1.26.

### CODE REVIEW

File: lib/codegen/build.rs [hunk 1]
```diff
@@ -18,12 +18,19 @@
...
 extern crate cranelift_codegen_meta as meta;
 
 use std::env;
 use std::process;
+use rustc_version::{version, Version};
 
 fn main() {
+    // Detect Rust 1.26.
+    if version().unwrap() >= Version::parse("1.26.0").unwrap() {
+        println!("cargo:rustc-cfg=rust_1_26");
```
Comment (lachlansneff) [inline lib/codegen/build.rs]: Could you rename the cfg option to `greater_than_rust_1_26` or `after_rust_1_26`?

Comment (sunfishcode) [reply lib/codegen/build.rs]: Done

**Comment by sunfishcode** [comment]
The build fails because Cargo.toml is still pulling in rustc_apfloat with Rust 1.25. I'm investigating.

**Comment by lachlansneff** [comment]
@sunfishcode Try making it an "optional" dependency.

**Comment by lachlansneff** [comment]
Any idea why it still isn't working?

**Pull request closed without merging**